core/fortran-example/tutorial/05_parallel_for/parallel_for_example.cpp

```diff
@@ -0,0 +1,244 @@
+/****************************************************************************
+ * Copyright (c) 2018-2019 by the Cabana authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the Cabana library. Cabana is distributed under a   *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <Cabana_Core.hpp>
+
+#include <Kokkos_Core.hpp>
+#include <Kokkos_Random.hpp>
+
+#include <iostream>
+
+/* Define the inner vector length of SOA */
+#include "veclen.h"
+
+//---------------------------------------------------------------------------//
+// parallel for example.
+//---------------------------------------------------------------------------//
+
+/* This is the coresponding struct_of_array defined by DataTypes (see below) */
+struct local_data_struct_t {     
+  double d0[veclen];     
+  double d1[veclen];     
+};
+
+/* Declare functions that will be mixed with Fortran */
+extern "C" {
+  /* written in C++; called by Fortran */
+  void c_kokkos_initlize( void ); 
+  void c_kokkos_finalize( void ); 
+  void parallelForExample();
+  
+  /* wirtten in Fortran; called by C++ */
+#if USE_GPU == 1  
```

Comment:
Can these be `__host__ __device__`?

Reply:
Yes, it can be. If __host__ is there, then the compiler will generate both the cpu  and gpu codes. Here we don't need it.

Reply:
I would still use `KOKKOS_FUNCTION` here as the decorator instead of these macros

Reply:
Done